### PR TITLE
Change "push" tests to be opt-in

### PR DESCRIPTION
**Pull request by cjllanwarne**
See the various commit test timings to see whether tests did or did not run.

Idea:
* One green light is as good as two since in any real situation the 'push' tests would be restarted if the 'PR' tests are green.
* We can still test on push if we really want to, using `FORCETEST` in the commit message
* We won't have so many "background" tests running at any given time, hopefully reducing the queue-time for PRs

In other words:
* Should significantly *increase* cycle time for PRs and *reduce* developer time shepherding PRs through two sets of identical tests

### CODE REVIEW

**Comment by cjllanwarne** [comment]
> More helpful FORCEPUSH docs when exiting early

🤦‍♂️ that's not `FORCETEST`

File: src/ci/bin/test.inc.sh [hunk 1]
```diff
@@ -138,6 +144,7 @@ cromwell::private::create_build_variables() {
             CROMWELL_BUILD_MYSQL_PASSWORD="${CROMWELL_BUILD_MYSQL_PASSWORD-}"
             CROMWELL_BUILD_MYSQL_SCHEMA="${CROMWELL_BUILD_MYSQL_SCHEMA-cromwell_test}"
             CROMWELL_BUILD_GENERATE_COVERAGE=true
+            CROMWELL_RUN_TESTS=true
```
Comment (mcovarr) [inline src/ci/bin/test.inc.sh]: could you DRY out everything by defaulting `CROMWELL_RUN_TESTS=true` and only setting it to `false` under the right circumstances?

**Comment by cjllanwarne** [comment]
@kshakir I just made the following changes:

- SBT is now run on pushes, to make sure that the artifact publishing still happens. Since it's ~30 minutes, not dependent on external services, and less flaky than the other tests I still think this is an improvement over today
- I switched the syntax to `[force ci]`. See the most recent commit message and it triggering the tests to run.

File: src/ci/bin/test.inc.sh [hunk 2]
```diff
@@ -747,6 +756,10 @@ cromwell::private::kill_tree() {
 
 cromwell::build::exec_test_script() {
     cromwell::private::create_build_variables
+    if [[ "${CROMWELL_RUN_TESTS}" == "false" ]]; then
+      echo "Use 'FORCETEST' in commit message to run tests on 'push'"
```
Comment (kshakir) [inline src/ci/bin/test.inc.sh]:
```suggestion
      echo "Use '[force ci]' in commit message to run tests on 'push'"
```

File: src/ci/bin/test.inc.sh [hunk 3]
```diff
@@ -79,6 +78,8 @@ cromwell::private::create_build_variables() {
       CROMWELL_BUILD_IS_VIRTUAL_ENV=false
     fi
 
+    CROMWELL_RUN_TESTS=true
```
Comment (kshakir) [inline src/ci/bin/test.inc.sh]:
```suggestion
    CROMWELL_BUILD_RUN_TESTS=true
```

All variables for the build are prefixed as `CROMWELL_BUILD_` to distinguish that the variable is affecting the **cromwell repository build**, versus just building the **cromwell server executable**.

File: src/ci/bin/test.inc.sh [hunk 4]
```diff
@@ -48,7 +48,6 @@ cromwell::private::create_build_variables() {
     else
         CROMWELL_BUILD_PROVIDER="${CROMWELL_BUILD_PROVIDER_UNKNOWN}"
     fi
-
```
Comment (kshakir) [inline src/ci/bin/test.inc.sh]: Revert?

File: src/ci/bin/test.inc.sh [hunk 5]
```diff
@@ -870,6 +883,8 @@ cromwell::build::publish_artifacts() {
     fi
 }
 
+
+
```
Comment (kshakir) [inline src/ci/bin/test.inc.sh]: Revert?
